easybuild/tools/module_generator.py

```diff
@@ -298,6 +299,100 @@ def use(self, paths, prefix=None, guarded=False):
+                else:
+                    text = sec_txt
+
+                return [
+                    '',
+                    '',
+                    sec_name,
+                    '=' * len(sec_name),
+                    text,
+                ]
+            else:
+                return []
+
+        # General package description (mandatory)
+        lines = generate_section('Description', self.app.cfg['description'], strip=True)
+
+        # Package usage instructions (optional)
+        lines.extend(generate_section('Usage', self.app.cfg['usage'], strip=True))
+
+        # Examples (optional)
+        lines.extend(generate_section('Examples', self.app.cfg['examples'], strip=True))
+
+        # Additional information: homepage + (if available) doc paths/urls, upstream/site contact
+        lines.extend(generate_section("More information", " - Homepage: %s" % self.app.cfg['homepage']))
+
+        docpaths = self.app.cfg['docpaths'] or []
+        docurls = self.app.cfg['docurls'] or []
+        if docpaths or docurls:
+            env_name = convert_name(self.app.name, upper=True)
+            root_envvar = ROOT_ENV_VAR_NAME_PREFIX + env_name
+            lines.extend([" - Documentation:"])
+            lines.extend(["    - $%s/%s" % (root_envvar,path) for path in docpaths])
+            lines.extend(["    - %s" % url for url in docurls])
+
+        upstream = self.app.cfg['upstream_contact']
+        if upstream:
+            if isinstance(upstream, list):
+                lines.extend([" - Upstream contacts:"])
+                lines.extend(["    - %s" % address for address in upstream])
+            else:
+                lines.extend([" - Upstream contact: %s" % upstream])
+
+        site_contact = self.app.cfg['site_contact']
+        if site_contact:
+            if isinstance(site_contact, list):
+                lines.extend([" - Site contacts:"])
+                lines.extend(["    - %s" % address for address in site_contact])
+            else:
+                lines.extend([" - Site contact: %s" % site_contact])
+
+        # Extensions (if any)
+        extensions = self._generate_extension_list()
+        lines.extend(generate_section("Included extensions", '\n'.join(wrap(extensions, 78))))
+
+        return '\n'.join(lines)
+
+    def _generate_whatis_lines(self):
+        """
+        Generate a list of entries used for `module whatis`.
+        """
+        whatis = self.app.cfg['whatis']
+        if whatis is None:
+            # default: include 'whatis' statements with description, homepage, and extensions (if any)
+            whatis = [
+                "Description: %s" % self.app.cfg['description'],
+                "Homepage: %s" % self.app.cfg['homepage']
+            ]
+            extensions = self._generate_extension_list()
+            if extensions:
+                whatis.append("Extensions: %s" % extensions)
+
+        return whatis
+
+    def _generate_extension_list(self):
```

Comment:
sort these methods alphabetically, let's not let @damianam's efforts go to waste ;)